http_test.go

```diff
@@ -109,12 +105,17 @@ func Test_EndToEnd(t *testing.T) {
 		})
 
 		Convey("Handling error conditions", func() {
-			Convey("When no page is specified", func() {
+			Convey("When the document is not written properly to disk", func() {
+				// Fetch a file which doesn't exist, but leave downloadShouldError = false
+				// so mockDownloader doesn't return an error.
 				req := httptest.NewRequest("GET", "/documents/somewhere/asdf.pdf", nil)
 				recorder := httptest.NewRecorder()
 
-				h.handleImage(recorder, req)
-				So(recorder.Result().StatusCode, ShouldEqual, 400)
+				h.handleDocument(recorder, req)
+				body, err := ioutil.ReadAll(recorder.Result().Body)
+				So(err, ShouldBeNil)
+				So(recorder.Result().StatusCode, ShouldEqual, 500)
```

Comment:
This means that something bad happened in our backend, since mupdf can't load the file we downloaded to disc. It could also mean we got a corrupted / bad file.